src/main/java/com/minecolonies/coremod/commands/CommandEntryPointNew.java

```diff
@@ -0,0 +1,596 @@
+            if (!currentMenuType.isNavigationMenu())
+            {
+                final ActionMenu actionMenu = (ActionMenu) menu;
+                for (final ActionArgument actionArgument : actionMenu.getActionArgumentList())
+                {
+                    sb.append(' ').append(actionArgument.getUsage());
+                }
+            }
+            return sb.toString();
+        }
+    }
+
+    private void buildParentPathString(final StringBuilder sb, final TreeNode<Menu> menuTreeNode)
+    {
+        final List<TreeNode<Menu>> treeNodeMenuListMinusRoot = new ArrayList<>();
+        TreeNode<Menu> currentMenuTreeNode = menuTreeNode;
+        while (null != currentMenuTreeNode.getParent())
+        {
+            treeNodeMenuListMinusRoot.add(currentMenuTreeNode);
+            currentMenuTreeNode = currentMenuTreeNode.getParent();
+        }
+        sb.append('/').append(currentMenuTreeNode.getData().getMenuItemName());
+
+        Collections.reverse(treeNodeMenuListMinusRoot);
+        for (final TreeNode<Menu> treeNode : treeNodeMenuListMinusRoot)
+        {
+            sb.append(' ').append(treeNode.getData().getMenuItemName());
+        }
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String[] args) throws CommandException
+    {
+        final BlockPos pos = null;
+        final ParsingResult parsingResult = getTabCompletionsAndParsingHolders(root, server, sender, args, pos);
+        final TreeNode<Menu> executionTreeNode = parsingResult.getExecutionTreeNode();
+        final List<ActionArgument> executionActionArgumentList = parsingResult.getExecutionActionArgumentList();
+        final String badArgument = parsingResult.getBadArgument();
+        if (null == executionTreeNode)
+        {
+            throw new CommandException(getCommandUsage(sender, root));
+        }
+
+        final Menu executionMenu = executionTreeNode.getData();
+        if (executionMenu.getMenuType().isNavigationMenu())
+        {
+            throw new CommandException(getCommandUsage(sender, executionTreeNode));
+        }
+
+        final ActionMenu actionMenu = (ActionMenu) executionMenu;
+
+        // Check that required arguments are provided with a valid value
+        final List<ActionArgument> actionArgumentListForActionMenu = actionMenu.getActionArgumentList();
+        for (final ActionArgument actionArgument : actionArgumentListForActionMenu)
+        {
+            if (actionArgument.isRequired())
+            {
+                boolean foundIt = false;
+                if (null != executionActionArgumentList)
+                {
+                    for (final ActionArgument executionActionArgument : executionActionArgumentList)
+                    {
+                        if (null != executionActionArgument)
+                        {
+                            if (actionArgument.getName().equals(executionActionArgument.getName()))
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 5 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)